app/src/main/java/com/example/tasky/agenda/data/local/AgendaDao.kt

```diff
@@ -0,0 +1,66 @@
+package com.example.tasky.agenda.data.local
+
+import androidx.room.Dao
+import androidx.room.Query
+import androidx.room.Upsert
+import com.example.tasky.agenda.data.local.model.EventEntity
+import com.example.tasky.agenda.data.local.model.ReminderEntity
+import com.example.tasky.agenda.data.local.model.TaskEntity
+
+@Dao
+interface AgendaDao {
+
+    @Upsert
+    suspend fun insertEvent(event: EventEntity)
+
+    @Query("DELETE FROM EventEntity WHERE id = :id")
+    suspend fun deleteEvent(id: String)
+
+    @Query("SELECT * FROM EventEntity WHERE id = :id")
+    suspend fun getEventById(id: String): EventEntity
+
+    @Query(
+        """
+            SELECT * FROM EventEntity
+            WHERE `from` >= :day
+            AND `from` < :endOfDay
+        """
+    )
+    suspend fun getEventsForGivenDay(day: Long, endOfDay: Long): List<EventEntity>
```

Comment:
When you get a list of items from Room, I recommend using Flows
Since that allows you to automatically refresh the UI without any modifications